spec/message.ebnf

```diff
@@ -1,25 +1,20 @@
-Message ::= Declaration* ( Pattern | Selector Variant+ )
+Message ::= (s? Declaration)* s? Body s?
 
-/* Preamble */
-Declaration ::= 'let' WhiteSpace Variable '=' '{' Expression '}'
-Selector ::= 'match' ( '{' Expression '}' )+
+Declaration ::= 'let' s Variable s? '=' s? '{' s? Expression s? '}'
+Body ::= Pattern | Selector (s? Variant)+
 
-/* Variants and Patterns */
-Variant ::= 'when' ( WhiteSpace VariantKey )+ Pattern
-VariantKey ::= Literal | Nmtoken | '*'
-Pattern ::= '{' (Text | Placeholder)* '}' /* ws: explicit */
+Pattern ::= '{' (Text | Placeholder)* '}'
```

Comment:
`Text` is defined such that backslashes in the text must be escaped, e.g. `\u12AB` is required to be `\\u12AB` or `\n` is required to be `\\n`. I'm okay with requiring `{` and `}` literals to be escaped.

I am nervous that we're recreating the apostrophe disaster in MF1 by requiring every backslash to be escaped. Developers use e.g. `\n` or other escapes in strings all the time. Making double escapes out of them doesn't seem to be required except to allow for the literal `\{` or `\}` to appear in a string. 

I think we might be better off by making the placeholder markers self-escaping, e.g. if you want to have a literal `{` use `{{`. `{` and `}` are rare enough in real text that this might be much less of an impact than requiring every backslash to be doubled.

Reply:
I filed #346 to discuss escaping independent of the whitespace.